scripts/aggregate-themes/aggregate-themes.js

```diff
@@ -0,0 +1,150 @@
+const fs = require('fs');
+const glob = require('glob');
+const path = require('path');
+const Logger = require('../utils/logger');
+
+const CONFIG = 'theme.config.js';
+const NODE_MODULES = 'node_modules/';
+const OUTPUT = 'aggregated-themes.js';
+const OUTPUT_PATH = path.resolve(process.cwd());
+const DISCLAIMER = fs.readFileSync(path.resolve(__dirname, 'disclaimer.txt'), 'utf8');
+
+/**
+ * Aggregates theme assets into a single file.
+ *
+ * By default the theme will recursively search all dependencies.
+ * If a theme directory contains a root-theme.scss only that single file will be aggregated.
+ */
+class ThemeAggregator {
+  /**
+   * Aggregates theme assets.
+   * @returns {string|null} - The output path of the aggregated theme file. Null if not generated.
+   */
+  static aggregate() {
+    const defaultConfig = path.resolve(process.cwd(), CONFIG);
+
+    if (fs.existsSync(defaultConfig)) {
+      // eslint-disable-next-line global-require, import/no-dynamic-require
+      return ThemeAggregator.aggregateTheme(require(defaultConfig));
+    }
+
+    return null;
+  }
+
+  /**
+   * Aggregates theme assets.
+   * @param {Object} options - The aggregation options.
+   * @returns {string} - The output path of the aggregated theme file.
+   */
+  static aggregateTheme(options = {}) {
+    ThemeAggregator.validate(options);
+
+    const { theme } = options;
+
+    const dirs = ThemeAggregator.find(`**/themes/${theme}/`, options);
```

Comment:
I think I'm a little lost. From what I can tell, we are filtering for files and directories it looks like initially, then we are filtering each returned dir/file to find `.css` files? We can pass `noDir` to glob to ensure we never return a directory.

It might be easier to match on these patters so you wouldn't need to filter on directories 
```
const rootStyles = ThemeAggregator.find(`**/themes/${theme}/root-theme.scss`, options);
const otherStyles = ThemeAggregator.find(`**/themes/${theme}/*.scss`, { exclude: options.exclude.push('root-theme.scss') });


// in filter method

then for each  path/root-theme.scss remove path/*.scss options from otherStyles list
```

Reply:
We are finding all theme directories and then filtering the contents of the directories. Not all theme directories will have a root-theme.scss file. If there is not a root-theme we add all the files from the directory. 

```
project
└── themes
    ├── terra-dark-theme
    │   ├── component.scss
    │   ├── root-theme.scss
    │   └── scoped-theme.scss
    └── terra-light-theme
        ├── component.scss // There is only a single file, there isn't a root-theme.scss
```

We only use the root-theme if it exists, otherwise we aggregate all available scss files in that directory.